lib/SPIRV/SPIRVToOCL.cpp

```diff
@@ -319,6 +323,41 @@ void SPIRVToOCL::visitCallSPIRVPipeBuiltin(CallInst *CI, Op OC) {
       &Attrs);
 }
 
+void SPIRVToOCL::visitCallSPIRVImageMediaBlockBuiltin(CallInst *CI, Op OC) {
+  AttributeList Attrs = CI->getCalledFunction()->getAttributes();
+  mutateCallInstOCL(
+      M, CI,
+      [=](CallInst *, std::vector<Value *> &Args) {
+        // Moving the first argument to the end.
+        std::rotate(Args.rbegin(), Args.rend() - 1, Args.rend());
+        Type *FuncType = CI->getType();
```

Comment:
```suggestion
        Type *RetType = CI->getType();
```

`FuncType` confuses me with `FunctionType` which is a bit different thing. I got quite a strange first impression trying to understand why are you expecting `FunctionType` to be equal to `VectorType`

Reply:
Done